### PR TITLE
Added initContainer in logs & terminal

### PR DESCRIPTION
**Pull request by milindk8**
# Description

On init container, logs are not getting shown on app detail page

Fixes # [ISSUE 498](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/498)

Screenshot 2022-09-20 at 2 28 59 PM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Create initContainer & deploy 

- By using terminal when initContainer is live
- By looking logs when it is live

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by vivek-devtron — APPROVED** [review]

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Logs.component.tsx [hunk 1]
```diff
@@ -715,7 +715,7 @@ function getInitialPodContainerSelection(
             };
         }
 
-        let containers = new Set(_allSelectedPods.flatMap((_po) => _po.containers ?? []));
+        let containers = new Set(_allSelectedPods.flatMap((_po) => flatContainers(_po) ?? []));
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Logs.component.tsx:718]: rename _po to something relevant like `_pod`

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Logs.component.tsx [hunk 2]
```diff
@@ -177,7 +178,7 @@ function LogsComponent({ selectedTab, isDeleted, logSearchTerms, setLogSearchTer
         let containers = podContainerOptions.containerOptions.filter((_co) => _co.selected).map((_co) => _co.name);
 
         let podsWithContainers = pods
-            .flatMap((_pod) => _pod.containers.map((_c) => [_pod.name, _c]))
+            .flatMap((_pod) => flatContainers(_pod).map((_c) => [_pod.name, _c]))
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Logs.component.tsx:181]: rename _c to `_container`

Comment (milindk8) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Logs.component.tsx]: Done

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Logs.component.tsx (hunk 1 above)
Comment (milindk8) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Logs.component.tsx]: Done

**Pull request merged**